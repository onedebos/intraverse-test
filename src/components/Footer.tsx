import React from "react";
import FooterComponent from "./FooterComponent";
import Button from "./LoremIpsumButton";

const Footer: React.FC = () => {
  return (
    <footer className="bg-blue-50 py-40 text-center">
      <div className="flex justify-center max-w-screen-lg m-auto text-left">
        <FooterComponent bgColor="bg-white" />
        <FooterComponent bgColor="bg-teal-1000" />
      </div>
      <h3 className="my-10">Lorem ipsum now?</h3>
      <Button />
    </footer>
  );
};

export default Footer;
