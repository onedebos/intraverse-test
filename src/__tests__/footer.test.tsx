import React from "react";
import { render } from "@testing-library/react";
import Footer from "../components/Footer";
import "@testing-library/jest-dom";

describe("Footer", () => {
  test("renders Footer component", () => {
    const tree = render(<Footer />);
    expect(tree).toMatchSnapshot();
    // screen.debug(); call this to see HTML tree
  });
});
