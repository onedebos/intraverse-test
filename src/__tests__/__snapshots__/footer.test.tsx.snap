// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="mt-40 md:mt-0 sub-bg-color p-10 w-full"
      >
        <div
          class="flex justify-evenly flex-col md:flex-row "
        >
          <div>
            <h2
              class="text-white font-bold text-3xl"
              data-testid="footer-logo"
            >
              TypeFastr
            </h2>
            <h4
              class="text-white text-sm"
            >
              © 2020, 
              <a
                href="https://adebola.dev"
              >
                Designed by Adebola.
              </a>
            </h4>
          </div>
          <div
            class="flex flex-col mt-4 md:mt-0"
          >
            <h1
              class="text-white font-bold"
            >
              About Creator
            </h1>
            <h3
              class="text-gray-500"
            >
              <a
                href="https://blog.adebola.dev"
              >
                Blog
              </a>
            </h3>
            <h3
              class="text-gray-500"
            >
              <a
                href="https://adebola.dev"
              >
                Portfolio
              </a>
            </h3>
          </div>
          <div
            class="flex flex-col mt-4 md:mt-0"
          >
            <h1
              class="text-white font-bold"
            >
              Contact
            </h1>
            <h3
              class="text-gray-500"
            >
              <a
                href="https://blog.adebola.dev"
              >
                Blog
              </a>
            </h3>
            <h3
              class="text-gray-500"
            >
              <a
                href="https://adebola.dev"
              >
                Portfolio
              </a>
            </h3>
          </div>
          <div
            class="flex flex-col mt-4 md:mt-0"
          >
            <h1
              class="text-white font-bold"
            >
              More
            </h1>
            <h3
              class="text-gray-500"
            >
              <a
                href="https://blog.adebola.dev"
              >
                TypeFastr
              </a>
            </h3>
            <h3
              class="text-gray-500"
            >
              <a
                href="https://filelockrr.com"
              >
                Other Products
              </a>
            </h3>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="mt-40 md:mt-0 sub-bg-color p-10 w-full"
    >
      <div
        class="flex justify-evenly flex-col md:flex-row "
      >
        <div>
          <h2
            class="text-white font-bold text-3xl"
            data-testid="footer-logo"
          >
            TypeFastr
          </h2>
          <h4
            class="text-white text-sm"
          >
            © 2020, 
            <a
              href="https://adebola.dev"
            >
              Designed by Adebola.
            </a>
          </h4>
        </div>
        <div
          class="flex flex-col mt-4 md:mt-0"
        >
          <h1
            class="text-white font-bold"
          >
            About Creator
          </h1>
          <h3
            class="text-gray-500"
          >
            <a
              href="https://blog.adebola.dev"
            >
              Blog
            </a>
          </h3>
          <h3
            class="text-gray-500"
          >
            <a
              href="https://adebola.dev"
            >
              Portfolio
            </a>
          </h3>
        </div>
        <div
          class="flex flex-col mt-4 md:mt-0"
        >
          <h1
            class="text-white font-bold"
          >
            Contact
          </h1>
          <h3
            class="text-gray-500"
          >
            <a
              href="https://blog.adebola.dev"
            >
              Blog
            </a>
          </h3>
          <h3
            class="text-gray-500"
          >
            <a
              href="https://adebola.dev"
            >
              Portfolio
            </a>
          </h3>
        </div>
        <div
          class="flex flex-col mt-4 md:mt-0"
        >
          <h1
            class="text-white font-bold"
          >
            More
          </h1>
          <h3
            class="text-gray-500"
          >
            <a
              href="https://blog.adebola.dev"
            >
              TypeFastr
            </a>
          </h3>
          <h3
            class="text-gray-500"
          >
            <a
              href="https://filelockrr.com"
            >
              Other Products
            </a>
          </h3>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
